dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java

```diff
@@ -471,85 +471,31 @@ else if (element.getName().equals("Context"))
                         {
-                            rp.setRpName(element.getAttributeValue("rpName"));
-                        }
-                        try {
-                            if (element.getAttributeValue("start-date") != null)
-                            {
-                                rp.setStartDate(sdf.parse(element.getAttributeValue("start-date")));
-                            }
-                            if (element.getAttributeValue("end-date") != null)
-                            {
-                                rp.setEndDate(sdf.parse(element.getAttributeValue("end-date")));
-                            }
-                        }catch (ParseException ex) {
-                            java.util.logging.Logger.getLogger(METSRightsCrosswalk.class.getName()).log(Level.SEVERE, null, ex);
+                        	rp.setEndDate(sdf.parse(element.getAttributeValue("end-date")));
                         }
-
-                        List<Element> le = new ArrayList<Element>(element.getChildren());
-                        for (Element el : le)
-                        {
-                            if ((el.getAttributeValue("DISCOVER").equalsIgnoreCase("true")) 
-                                    && (el.getAttributeValue("DISPLAY").equalsIgnoreCase("true")))
-                            {
-                                if (el.getAttributeValue("DELETE").equalsIgnoreCase("false"))
-                                {
-                                    if (el.getAttributeValue("MODIFY").equalsIgnoreCase("false"))
-                                    {
-                                        rp.setAction(Constants.READ);
-                                    }
-                                    else
-                                    {
-                                        rp.setAction(Constants.WRITE);
-                                    }
-                                }
-                                else
-                                {
-                                    if (el.getAttributeValue("MODIFY").equalsIgnoreCase("true"))
-                                    {
-                                        rp.setAction(Constants.DELETE);
-                                        if ((el.getAttributeValue("COPY").equalsIgnoreCase("true"))
-                                                &&(el.getAttributeValue("DUPLICATE").equalsIgnoreCase("true"))
-                                                &&(el.getAttributeValue("PRINT").equalsIgnoreCase("true")))
-                                        {
-                                            rp.setAction(Constants.ADMIN);
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        policies.add(rp);
-                    } catch (NullPointerException ex) {
```

Comment:
Hi @mjmarttila : I understand now. Thanks for the clear explanation

Technically, according to the [METSRights XSD](https://www.loc.gov/standards/rights/METSRights.xsd), a `<Context>` element can have multiple `<Permissions>` elements. (See the `<xs:complexType name="ContextTYPE">` in that XSD).

However, I have never seen an example of why this would be.  Even the examples provided for METSRights only ever have one: http://www.loc.gov/standards/rights/

So, considering this option works and is more simplistic, it seems reasonable enough. I don't know why you'd ever want more than one of these `<Permissions>` tags.

However, I think it would be worth adding comments to this effect in your changes... So, perhaps adding to line 480 a comment like `// Note: we are assuming here there will only ever be ONE <Permissions> element. Currently there a no known use cases for multiple.`